src/signalrclient/CMakeLists.txt

```diff
@@ -1,26 +1,34 @@
 
 
 set (SOURCES
- callback_manager.cpp
- connection.cpp
- connection_impl.cpp
- default_websocket_client.cpp
- http_sender.cpp
- hub_connection.cpp
- hub_connection_impl.cpp
- logger.cpp
- request_sender.cpp
- signalr_client_config.cpp
- stdafx.cpp
- trace_log_writer.cpp
- transport.cpp
- transport_factory.cpp
- url_builder.cpp
- web_request.cpp
- web_request_factory.cpp
- websocket_transport.cpp
+  callback_manager.cpp
+  connection.cpp
+  connection_impl.cpp
+  default_http_client.cpp
+  default_websocket_client.cpp
+  http_sender.cpp
+  hub_connection.cpp
+  hub_connection_impl.cpp
+  logger.cpp
+  negotiate.cpp
+  signalr_client_config.cpp
+  stdafx.cpp
+  trace_log_writer.cpp
+  transport.cpp
+  transport_factory.cpp
+  url_builder.cpp
+  web_request.cpp
+  web_request_factory.cpp
+  websocket_transport.cpp
 )
 
 add_library (signalrclient SHARED ${SOURCES})
 
-target_link_libraries(signalrclient ${CPPREST_SO})
+if(NOT WIN32)
+  target_link_libraries(signalrclient ${CPPREST_LIB} ${OPENSSL_LIBRARIES} Boost::boost Boost::system Boost::chrono Boost::thread)
```

Comment:
If you're using CMake 3.4 or higher, OpenSSL has nice target names[1]: `OpenSSL::SSL` and `OpenSSL::Crypto`.

[1] https://cmake.org/cmake/help/v3.4/module/FindOpenSSL.html

Reply:
You should also consider marking these as `PUBLIC` or `PRIVATE` links, based on whether you include them in your own public headers.

https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents